spec/models/supporting_evidence_attachment_spec.rb

```diff
@@ -0,0 +1,121 @@
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile.pdf' }.to_json
+          )
+        end
+
+        it 'obscures all but the first 3 and last 2 characters of the filename' do
+          expect(attachment.obscured_filename).to eq('MyS*******le.pdf')
+        end
+      end
+
+      context 'for a filename containing numbers' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile123.pdf' }.to_json
+          )
+        end
+
+        it 'obscures the filename' do
+          expect(attachment.obscured_filename).to eq('MyS**********23.pdf')
+        end
+      end
+
+      context 'for a filename containing underscores' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile_123.pdf' }.to_json
+          )
+        end
+
+        it 'does not obscure the underscores' do
+          expect(attachment.obscured_filename).to eq('MyS*********_*23.pdf')
+        end
+      end
+
+      context 'for a filename containing dashes' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile-123.pdf' }.to_json
+          )
+        end
+
+        it 'does not obscure the dashes' do
+          expect(attachment.obscured_filename).to eq('MyS*********-*23.pdf')
+        end
+      end
+
+      # Ensure Regexp still handles the file extension properly
+      context 'for a filename containing a dot' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecret.File123.pdf' }.to_json
+          )
+        end
+
+        it 'obsucres the filename properly' do
+          expect(attachment.obscured_filename).to eq('MyS*****.*****23.pdf')
+        end
+      end
+
+      context 'for a filename containing whitespace' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'My Secret File.pdf' }.to_json
+          )
+        end
+
+        it 'preserves the whitespace' do
+          expect(attachment.obscured_filename).to eq('My ****** **le.pdf')
+        end
+      end
+    end
+
+    context 'for a filename with five characters or less' do
+      let(:attachment) do
+        build(
+          :supporting_evidence_attachment,
+          file_data: { filename: 'File.pdf' }.to_json
+        )
+      end
+
+      it 'does not obscure the filename' do
+        expect(attachment.obscured_filename).to eq('File.pdf')
+      end
+    end
+
+    # NOTE: Filetypes that need to be converted in EVSSClaimDocumentUploaderBase
+    # have 'converted_' appended to the file name and saved as convered_filename in the file_data
+    # Ensure the obscured_filename method is masking the orignal filename
```

Comment:
Typos
- `convered_filename`
- `orignal`

'appended' should be 'prepended'

Reply:
Fixed in https://github.com/department-of-veterans-affairs/vets-api/pull/16359/commits/38c7f164554b917667b82821e69302baee9a569d